Remotion/Data/DomainObjects.UnitTests/QueriesForStandardMapping.xml

```diff
@@ -138,30 +138,47 @@
     </statement>
   </query>
 
-  <query id="StoredProcedureQuery" type="collection">
+  <query id="StoredProcedureQuery" type="collection-readonly">
     <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
     <statement>rpf_testSPQuery</statement>
     <collectionType>Remotion.Data.DomainObjects.UnitTests.TestDomain.OrderCollection, Remotion.Data.DomainObjects.UnitTests</collectionType>
   </query>
 
-  <query id="StoredProcedureQueryWithParameter" type="collection">
+  <query id="StoredProcedureQueryWithParameter" type="collection-readonly">
     <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
     <statement>exec rpf_testSPQueryWithParameter @customerID</statement>
     <collectionType>Remotion.Data.DomainObjects.UnitTests.TestDomain.OrderCollection, Remotion.Data.DomainObjects.UnitTests</collectionType>
   </query>
 
-  <query id="FakeQuery" type="collection">
+  <query id="FakeQuery" type="collection-readonly">
     <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
     <statement>SELECT "Dummy"</statement>
     <collectionType>Remotion.Data.DomainObjects::DomainObjectCollection</collectionType>
   </query>
 
-  <query id="CustomQuery" type="custom">
+  <query id="CollectionQuery" type="collection-readonly">
+    <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
+    <statement>SELECT "Dummy"</statement>
+    <collectionType>Remotion.Data.DomainObjects::DomainObjectCollection</collectionType>
+  </query>
+
+  <query id="CollectionQueryReadWrite" type="collection-readwrite">
+    <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
+    <statement>SELECT "Dummy"</statement>
+    <collectionType>Remotion.Data.DomainObjects::DomainObjectCollection</collectionType>
+  </query>
+
+  <query id="CustomQuery" type="custom-readonly">
+    <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
+    <statement>SELECT String, Int16, Boolean, Enum, ExtensibleEnum  FROM [TableWithAllDataTypes]</statement>
+  </query>
+
+  <query id="CustomQueryReadWrite" type="custom-readwrite">
```

Comment:
Question since it's difficult to see via review: do we actually use those new queris in the tests and if so, where?

Reply:
Im not sure what the exact file name was, but we use them for one of the new argument checks for each of the query types